internal/controller/controllers/agent_controller.go

```diff
@@ -905,6 +915,50 @@ func (r *AgentReconciler) updateIfNeeded(ctx context.Context, log logrus.FieldLo
 	return nil
 }
 
+func (r *AgentReconciler) getIgnitionToken(ctx context.Context, log logrus.FieldLogger, agent *aiv1beta1.Agent) (string, error) {
+	ignitionEndpointTokenReference := agent.Spec.IgnitionEndpointTokenReference
+
+	if ignitionEndpointTokenReference == nil {
+		log.Info("No IgnitionEndpointTokenReference set")
+		return "", errors.New("No IgnitionEndpointTokenReference set")
+	}
+
+	secret := &corev1.Secret{}
+	secretRef := types.NamespacedName{Namespace: ignitionEndpointTokenReference.Namespace, Name: ignitionEndpointTokenReference.Name}
+	if err := r.Get(ctx, secretRef, secret); err != nil {
+		log.WithError(err).Errorf("Failed to get user-data secret %s", ignitionEndpointTokenReference.Name)
+		return "", err
+	}
+
+	ignitionConfig := &ignitionapi.Config{}
+	if err := json.Unmarshal(secret.Data["value"], ignitionConfig); err != nil {
```

Comment:
I don't understand what this secret holds. A whole ignition? why not the Bearer token only?
Need to add documentation.

Reply:
Yes, a whole ignition.  HyperShift adds it to the Machine, which gets passed to the AgentMachine and finally to the Agent.

Reply:
.